src/CMakeLists.txt

```diff
@@ -2,41 +2,28 @@
 #
 # Copyright (c) 2021-2022 Alexander Kurbatov
 
-if (BUILD_FOR_LADDER)
-    ADD_DEFINITIONS(-DBUILD_FOR_LADDER)
-endif ()
-
-# All the source files for the bot.
-file(GLOB_RECURSE BOT_SOURCES "*.cpp" "*.h")
+set(bot_sources
+    main.cpp
+    Bot.cpp)
 
-include_directories(SYSTEM
-    ${PROJECT_SOURCE_DIR}/contrib/cpp-sc2/include
-    ${PROJECT_SOURCE_DIR}/contrib/cpp-sc2/contrib/protobuf/src
-    ${PROJECT_BINARY_DIR}/contrib/cpp-sc2/generated
-    "."
-)
+add_executable(BlankBot ${bot_sources})
 
-link_directories(${PROJECT_BINARY_DIR}/contrib/cpp-sc2/bin)
+if (BUILD_FOR_LADDER)
+    target_compile_definitions(BlankBot PRIVATE BUILD_FOR_LADDER)
+endif ()
 
 if (MSVC)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /EHsc")
+    target_compile_options(BlankBot PRIVATE /W4 /EHsc)
 else ()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic")
-endif ()
+    target_compile_options(BlankBot PRIVATE -Wall -Wextra -pedantic)
+endif()
 
-# Create the executable.
-add_executable(BlankBot ${BOT_SOURCES})
 target_link_libraries(BlankBot PRIVATE cpp_sc2)
 
 if (MINGW)
     target_link_libraries(BlankBot PRIVATE ssp)
-endif ()
-
-if (APPLE)
+elseif (APPLE)
     target_link_libraries(BlankBot PRIVATE "-framework Carbon")
-endif ()
-
-# Linux specific.
-if (UNIX AND NOT APPLE)
+elseif (UNIT AND NOT APPLE)
```

Comment:
Typo: `UNIT` -> `UNIX`.

Also:
* If we use `elseif` there is no need to doublecheck `NOT APPLE`.
* Please keep the original comment regarding Linux, it is kept mostly for people less experienced with cmake.